integration_tests/src/main/python/data_gen.py

```diff
@@ -208,6 +209,37 @@ def __init__(self, nullable=True, min_val = INT_MIN, max_val = INT_MAX,
     def start(self, rand):
         self._start(rand, lambda : rand.randint(self._min_val, self._max_val))
 
+
+#Maximum number of decimal digits a Long can represent is 18.
+DECIMAL_MIN = -(1 << 59)
+DECIMAL_MAX = (1 << 59)  #approximate
+class DecimalGen(DataGen):
+    """Generate Decimals, with some built in corner cases."""
+    def __init__(self, precision=None, scale=None, nullable=True, special_cases=[]):
+        if precision is None:
+            precision = 18
+            scale = 0
+            special_cases = [Decimal('-1'), Decimal('0'), Decimal('1'), Decimal(DECIMAL_MIN), Decimal(DECIMAL_MAX)]
+        super().__init__(DecimalType(precision, scale), nullable=nullable, special_cases=special_cases)
+        self._scale = scale
+        self._precision = precision
+        if (scale > 0):
+            pattern = "[0-9]{1,"+ str(precision - scale) + "}\.[0-9]{0," + str(scale) + "}"
```

Comment:
I made a mistake here.  If scale > precision, then this pattern is not going to work.  I think we might just want to use the other pattern all of the time.

Reply:
done